crates/symbolicli/src/main.rs

```diff
@@ -0,0 +1,655 @@
+use std::path::Path;
+use std::sync::Arc;
+
+use symbolic::common::split_path;
+use symbolicator_service::services::symbolication::SymbolicationActor;
+use symbolicator_service::types::{CompletedSymbolicationResponse, FrameTrust, Scope};
+use symbolicator_sources::{SentrySourceConfig, SourceConfig, SourceId};
+
+use anyhow::Context;
+use prettytable::format::consts::FORMAT_CLEAN;
+use prettytable::{cell, row, Row, Table};
+use reqwest::{header, Url};
+use tempfile::{NamedTempFile, TempPath};
+
+mod settings;
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let settings::Settings {
+        event_id,
+        project,
+        org,
+        auth_token,
+        base_url,
+        symbolicator_config,
+        output_format,
+    } = settings::Settings::get()?;
+
+    tracing_subscriber::fmt::init();
+
+    let runtime = tokio::runtime::Handle::current();
+    let (symbolication, _objects) =
+        symbolicator_service::services::create_service(&symbolicator_config, runtime)
+            .await
+            .context("failed to start symbolication service")?;
+
+    let mut headers = header::HeaderMap::new();
+    headers.insert(
+        header::AUTHORIZATION,
+        header::HeaderValue::from_str(&format!("Bearer {auth_token}")).unwrap(),
+    );
+
+    let client = reqwest::Client::builder()
+        .default_headers(headers)
+        .build()
+        .unwrap();
+
+    let project_source = SourceConfig::Sentry(Arc::new(SentrySourceConfig {
+        id: SourceId::new("sentry:project"),
+        token: auth_token,
+        url: base_url
+            .join(&format!("projects/{org}/{project}/files/dsyms/"))
+            .unwrap(),
+    }));
+    let mut sources = vec![project_source.clone()];
+    sources.extend(symbolicator_config.sources.iter().cloned());
+    let sources = Arc::from(sources.into_boxed_slice());
+
+    let scope = Scope::Scoped(project.clone());
+
+    let res = match Payload::parse(&event_id)? {
+        Some(local) => {
+            tracing::info!("successfully parsed local event");
+            local.process(&symbolication, scope, sources).await?
+        }
+        None => {
+            tracing::info!("event not found in local file system");
+            let remote = Payload::get_remote(&client, &base_url, &org, &project, &event_id).await?;
+            remote.process(&symbolication, scope, sources).await?
+        }
+    };
+
+    match output_format {
+        settings::OutputFormat::Json => println!("{}", serde_json::to_string(&res).unwrap()),
+        settings::OutputFormat::Compact => print_compact(res),
+        settings::OutputFormat::Pretty => print_pretty(res),
+    }
+
+    Ok(())
+}
+
+fn print_compact(mut response: CompletedSymbolicationResponse) {
+    let module_addr_by_code_file: HashMap<_, _> = response
+        .modules
+        .into_iter()
+        .filter_map(|module| Some((module.raw.code_file.clone()?, module)))
+        .collect();
+
+    if response.stacktraces.is_empty() {
```

Comment:
We can move it above `module_addr_by_code_file` to not collect them unnecessarily